### PR TITLE
Feat/68 more e2e tests

### LINKED ISSUE #68: Improve vue Integration tests
Add...

- [x] click checkbox and close popup
- [x] annotate without closing popup does not work
- [x] clicking button stores annotation
- [x] next image is shown once image has been annotated

### PR DESCRIPTION
**Pull request by bewee**
Based on #79 

### Changes made by this Pull Request

- Adds improved vue integration tests

### Definition of Done

- [x] Tests are written and run successful
- [x] Code is documented
- [x] User Manual is updated
- [x] All requirements of the user story are met

### CODE REVIEW

**Review by jdsauer — APPROVED** [review]
feels much more clean after changing the tests and deleting duplicate tests

File: static/cypress/integration/utils.ts [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+    { fixture: "task.json" }
+  ).as("get_task");
+}
+
+export function intercept_next_image() {
+  cy.intercept(
+    "GET",
+    `${Cypress.env("API_URL")}/tasks/ecg-qrs-classification-physiodb/next`,
+    {
+      fixture: "nextImage.txt",
```
Comment (jb3rndt) [inline static/cypress/integration/utils.ts:39]: Should this maybe return the sloth.jpg as we have it here now too?

File: static/cypress/integration/utils.ts [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+  config["intercept_login"] ||= intercept_login;
+  config["intercept_get_task"] ||= intercept_get_task;
+  config["intercept_next_image"] ||= intercept_next_image;
+  config["intercept_get_image"] ||= intercept_get_image;
+  config["intercept_store_annotation"] ||= intercept_store_annotation;
+  config["intercept_login"]();
+  config["intercept_get_task"]();
+  config["intercept_next_image"]();
+  config["intercept_get_image"]();
+  config["intercept_store_annotation"]();
```
Comment (jb3rndt) [inline static/cypress/integration/utils.ts]: I think we can consider putting all intercepts into a separate intercept function that handles this logic. It would have to be called in every test then too, but it would be clearer that we are mocking the whole api away. What do you think?

Apart from that I think it would be nice to have some kind of default intercept config like 
```js
defaultInterceptConfig = {
  intercept_login: intercept_login,
  ...
}

export function login(interceptConfig: { [key: string]: () => void } = {}) {
  const interceptConfig = { ...defaultInterceptConfig, ...interceptConfig };
  ...
}
```

File: static/cypress/support/commands.js [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+// -- This is a child command --
+// Cypress.Commands.add('drag', { prevSubject: 'element'}, (subject, options) => { ... })
+//
+//
+// -- This is a dual command --
+// Cypress.Commands.add('dismiss', { prevSubject: 'optional'}, (subject, options) => { ... })
+//
+//
+// -- This will overwrite an existing command --
+// Cypress.Commands.overwrite('visit', (originalFn, url, options) => { ... })
```
Comment (jb3rndt) [inline static/cypress/support/commands.js]: Is this file created by accident? I think we already have the typescript version of it

File: static/cypress/integration/utils.ts [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+    cy
+      .spy(
+        {
+          handle: (req: CyHttpMessages.IncomingHttpRequest) => {
+            req.reply({ statusCode: 204 });
+          },
+        },
+        "handle"
+      )
+      .as("store_annotation_spy")
```
Comment (jb3rndt) [inline static/cypress/integration/utils.ts]: What does this spy thing do? It seems to be only in this intercept

**Review by jb3rndt — COMMENTED** [review]
Looks nice, it feels good to have some more frontend tests 👍

File: static/cypress/support/commands.js (hunk 3 above)
Comment (bewee) [reply static/cypress/support/commands.js]: totally

File: static/cypress/integration/utils.ts (hunk 2 above)
Comment (bewee) [reply static/cypress/integration/utils.ts]: Yeah, it's a little dirty to solve it using the login function. On the other hand, we will most likely start every single test with calling the login method anyway... Really unsure about this tbh
